### DISCUSSION TITLE
Subscribe to existing PoseStamped topic and send to RobotStudio

### DISCUSSION THREAD

**Discussion by sharpworks-dev** [ros-industrial/abb_libegm#160] [Q&A]
Hello, this is my first post ever and I am very new to ROS so apologies if I am missing any obvious info.

I am working on a ROS project where I have a Unity scene set up with a [ROS-TCP-Connector](https://github.com/Unity-Technologies/ROS-TCP-Connector) that is sending PoseStamped messages on a topic I've called "teleop". These messages are received by an Ubuntu virtual machine using [ROS-TCP-Endpoint](https://github.com/Unity-Technologies/ROS-TCP-Endpoint) on the same computer via a bridged connection.

I would now like to use this package to subscribe to this topic, receive those PoseStamped messages, and send them to RobotStudio to control our robot there. As I said, I am very new to ROS and also unfamiliar with C++, so I tried looking at the code samples provided  [here](https://github.com/ros-industrial/abb_libegm/issues/18#issuecomment-457592204) but I just can't wrap my head around them.

Could someone guide me on how to modify the nodes or write a new node that can accomplish this?

**Comment by gavanderhoorn** [comment]
I would perhaps suggest to post this on the [abb_robot_driver](https://github.com/ros-industrial/abb_robot_driver) discussion board.

EGM is a very low-level interface to the controller, and while what you describe could be implemented with it, seeing as you are already using ROS, you might instead want to reuse `abb_robot_driver`.

Note: you write you have `PoseStamped` messages. You will have to put those through an IK solver to get the corresponding joint space poses. You could then publish those joint space poses to an instance of the `ros_control` `JointGroupPositionController`.

Note 2: requirements on motion smoothness, continuity and collision checking will be for you to satisfy. If you're going with the `JointGroupPositionController`, it will not do anything "smart", so be aware of that.
